### ISSUE TITLE
windows 'dlv debug' include 'exec.LookPath(os.Args[0])' error

### ISSUE THREAD

**Issue by tupunco** [go-delve/delve#505]
1. What version of Delve are you using (`dlv version`)?

`master`
1. What version of Go are you using? (`go version`)?

`1.6/windows`
1. What operating system and processor architecture are you using?

`windows 10`
1. What did you do?

`dlv debug`
1. What did you expect to see?

`current exe path`
1. What did you see instead?

`error`
# Detailed instructions

**Demo code**

``` go
package main

import (
    "fmt"
    "os"
    "os/exec"
    "path/filepath"
)

// execPath returns the executable path.
func execPath() (string, error) {
    filePath := os.Args[0]
    file, err := exec.LookPath(filePath)
    if err != nil {
        return "", err
    }
    return filepath.Abs(file)
}

func main() {
    path, err := execPath()
    fmt.Printf("%s-%s", path, err)
}
```

**Problem code**

``` go
exec.LookPath(os.Args[0])
```

**Problem analysis**
On Windows `exec.LookPath(path)`  function implementation to determine whether the `path` for `exe/com /..` File extension. Call the `dlv debug` command will generate a `debug` file , it no file extension.

**Solve the problem**
Modify the generated logic of the `debug` file, add the `exe` extension.
A simple solution, modify [commands.go#L50](https://github.com/derekparker/delve/blob/master/cmd/dlv/cmds/commands.go#L50) to :

``` go
var (
    debugname     = "debug"
    testdebugname = "debug.test"
)

func init() {
    switch os := runtime.GOOS; os {
    case "windows":
        debugname = "debug.exe"
        testdebugname = "debug.test.exe"
    }
}
```

`Note: English is poor, hope can understand`

**Comment by aarzilli** [comment]
Sounds reasonable, You can make a Pull Request if you want, if you do include a regression test.

**Comment by aarzilli** [comment]
Fixed by #1841
